internal/controller/rhdh/backstage.go

```diff
@@ -36,121 +34,161 @@ var ConfigMapNameAndConfigDataKey = map[string]string{
 
-	if err := client.Get(ctx, types.NamespacedName{
-		Namespace: operator.Subscription.TargetNamespace,
-		Name:      BackstageCRName,
-	}, &rhdh.Backstage{}); apierrors.IsNotFound(err) {
-		secret := rhdh.ObjectKeyRef{
-			Name: operator.SecretRef.Name,
-		}
-		backstageCR := &rhdh.Backstage{
-			TypeMeta: metav1.TypeMeta{
-				APIVersion: BackstageAPIVersion,
-				Kind:       BackstageKind,
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      BackstageCRName,
-				Namespace: operator.Subscription.TargetNamespace,
-				Labels:    operations.AddLabel(),
-			},
-			Spec: rhdh.BackstageSpec{
-				Application: &rhdh.Application{
-					AppConfig:                   &rhdh.AppConfig{ConfigMaps: bsConfigMapList},
-					DynamicPluginsConfigMapName: AppConfigRHDHDynamicPluginName,
-					ExtraEnvs: &rhdh.ExtraEnvs{
-						Secrets: []rhdh.ObjectKeyRef{secret},
+	rhdhNamespace := rhdhConfig.RHDHNamespace
+	rhdhName := rhdhConfig.RHDHName
+
+	if err := client.Get(ctx, types.NamespacedName{Namespace: rhdhNamespace, Name: rhdhName}, &rhdhv1alpha2.Backstage{}); err != nil {
+		if apierrors.IsNotFound(err) {
+			secret := rhdhv1alpha2.ObjectKeyRef{
+				Name: BackendAuthSecretName,
+			}
+			backstageCR := &rhdhv1alpha2.Backstage{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: BackstageAPIVersion,
+					Kind:       BackstageKind,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      BackstageCRName,
+					Namespace: rhdhConfig.RHDHNamespace,
+					Labels:    kubeoperations.AddLabel(),
+				},
+				Spec: rhdhv1alpha2.BackstageSpec{
+					Application: &rhdhv1alpha2.Application{
+						AppConfig:                   &rhdhv1alpha2.AppConfig{ConfigMaps: bsConfigMapList},
+						DynamicPluginsConfigMapName: AppConfigRHDHDynamicPluginName,
+						ExtraEnvs: &rhdhv1alpha2.ExtraEnvs{
+							Secrets: []rhdhv1alpha2.ObjectKeyRef{secret},
+						},
+						Replicas: util.MakePointer(BackstageReplica),
 					},
-					Replicas: util.MakePointer(BackstageReplica),
 				},
-			},
-		}
-		if err := client.Create(ctx, backstageCR); err != nil {
-			bsLogger.Error(err, "Error occurred when creating Backstage resource")
-			return err
+			}
+			if err := client.Create(ctx, backstageCR); err != nil {
+				rhdhLogger.Error(err, "Error occurred when creating Backstage resource", "CR-Name", rhdhName)
+				return err
+			}
+			rhdhLogger.Info("Successfully created Backstage resource", "CR-Name", rhdhName)
 		}
-		bsLogger.Info("Successfully created Backstage resource")
+		rhdhLogger.Error(err, "Error occurred when creating Backstage resource", "CR-Name", rhdhName)
+		return err
 	}
 	return nil
 }
 
-func GetConfigmapList(ctx context.Context, client client.Client, clusterDomain string,
-	operator orchestratorv1alpha1.RHDHOperator,
-	rhdhPlugins orchestratorv1alpha1.RHDHPlugins) []rhdh.ObjectKeyRef {
+func GetConfigmapList(ctx context.Context, client client.Client,
+	clusterDomain, wfNamespace string,
```

Comment:
I think you renamed `wfNamespace` and other `wf` variables elsewhere, you may have missed this one

Reply:
Updated those now. fingers crossed, didn't miss any.